packages/synapse-react-client/src/components/DataGrid/useDataGridWebsocket.ts

```diff
@@ -23,10 +23,24 @@ export function useDataGridWebSocket() {
     replicaId: number
     sessionId: string
   } | null>(null)
+  // Track the last connection parameters to detect changes - use ref to avoid callback recreation
```

Comment:
track the session info to determine when a model should be kept (when connecting to exisiting session) or cleared (connecting to a new session)